mantis-runtime/src/main/java/io/mantisrx/runtime/parameter/Parameters.java

```diff
@@ -38,22 +37,34 @@ public Parameters(Map<String, Object> state,
         this.parameterDefinitions = parameterDefinitions;
     }
 
+    /**
+     * Get parameter value given key with validation.
+     *
+     * If the key is required, parameters must have a value provided otherwise it will throw an exception.
+     * If the key is not defined, it will throw an exception.
+     */
     public Object get(String key) {
-        // check if required parameter, make sure
-        // has a value
+        // check if required parameter, make sure has a value
         if (requiredParameters.contains(key) &&
                 !state.containsKey(key)) {
-            throw new ParameterException("Attempting to reference a required parameter witn no value: " + key + ", check parameter definitions for job.");
+            throw new ParameterException("Attempting to reference a required parameter witn no value: " + key +
+                ", check parameter definitions for job.");
         }
         // check if parameter definition exists
         if (!parameterDefinitions.contains(key)) {
-            throw new ParameterException("Attempting to reference parameter: " + key + ", with no definition, check parameter definitions in job.");
+            throw new ParameterException("Attempting to reference parameter: " + key +
+                ", with no definition, check parameter definitions in job.");
         }
         return state.get(key);
     }
 
+    /**
+     * Get parameter value given key without validation.
+     *
+     * If the key is not defined or missing a provided value, the given default value will be returned.
+     */
     public Object get(String key, Object defaultValue) {
-        Object value = defaultValue;
```

Comment:
Do the semantics actually change with this? My initial read of the code says no, but if it does it may unintentionally impact other jobs.

Reply:
No, all branches will assign value so it's redundant to have an initial value.

Reply:
But it's cosmetic and I can revert that. Or better I can rewrite the whole function to use multiple returns.
```
        try {
            final Object value = get(key);
            return value != null ? value : defaultValue;
        } catch (ParameterException ex) {
            return defaultValue;
        }
```